Sources/ArgumentParser/Usage/HelpGenerator.swift

```diff
@@ -18,19 +18,72 @@ internal struct HelpGenerator {
     struct Element: Hashable {
       var label: String
       var abstract: String = ""
-      var discussion: String = ""
-      
+      var discussion: Discussion = .staticString("")
+
+      enum Discussion: Hashable {
+        case staticString(String)
+        case enumerable([Option])
+
+        struct Option: Hashable {
+          var name: String
+          var help: String
+        }
+
+        public init(_ discussion: ArgumentDefinition.Help.Discussion) {
+          switch discussion {
+          case .staticString(let s):
+            self = .staticString(s)
+          case .enumerated(let type):
+            let values = (type.allCases as? [any EnumerableOption]) ?? []
+            // TODO: to format
+            self = .enumerable(values.compactMap({ Option(name: $0.name, help: $0.help?.abstract ?? "")}))
+          }
+        }
+      }
+
       var paddedLabel: String {
         String(repeating: " ", count: HelpGenerator.helpIndent) + label
       }
-      
+
+      // TODO: to format
       func rendered(screenWidth: Int) -> String {
         let paddedLabel = self.paddedLabel
         let wrappedAbstract = self.abstract
           .wrapped(to: screenWidth, wrappingIndent: HelpGenerator.labelColumnWidth)
-        let wrappedDiscussion = self.discussion.isEmpty
+
+        var wrappedDiscussion = ""
+
+        if case let .staticString(discussionText) = discussion {
```

Comment:
thought: If discussion is optional then this can be a regular if-left.

Reply:
I may keep this as is just because I'll have to access the associated values for each case - did you have another approach in mind? :)